### PR TITLE
*: add new region score formula

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Make region score grow smoother and less affected by available changes.

### What is changed and how it works?
- introduce new formula (v2)
- add config to control formula version
- add test

Default formula for newly created clusters is "v2", all old clusters keeps "v1" unless set manually.

The test result shows that compared to v4.0.7, the new formula can help balance fast to converge and more stable even with a very high store limit configuration.

###  Check List

Tests
- Unit test

### Related Issue
- https://github.com/pingcap/tidb/issues/18012

### Release note
- Introduce new region score formula

### CODE REVIEW

File: server/core/store.go [hunk 1]
```diff
@@ -279,6 +294,26 @@ func (s *StoreInfo) RegionScore(highSpaceRatio, lowSpaceRatio float64, delta int
...
+	var (
+		K, M float64 = 1, 256 // Experience value to control the weight of the available influence on score
+		F    float64 = 20     // Experience value to prevent some nodes from running out of disk space prematurely.
+	)
+
+	var score float64
+	if A >= C || C < 1 {
+		score = R
+	} else if A > F {
+		score = (K + M*(math.Log(C)-math.Log(A-F+1))/(C-A+F-1)) * R
```
Comment (rleungx) [inline server/core/store.go:310]: How about adding some comments or some reference to explain why the function look like this?

Comment (disksing) [reply server/core/store.go:310]: good idea.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — COMMENTED** [review]
LGTM, do we need to add some test result in this pr desc?

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by disksing** [comment]
> LGTM, do we need to add some test result in this pr desc?

ok, I'll add a short description.

**Pull request merged**